### PR TITLE
Add synchronization attributes

### PR DESCRIPTION
**Pull request by cleclefibanity**

### CODE REVIEW

**Comment by cleclefibanity** [comment]
New synchronization attribute: customerOnline true|false (false if not set), and customerIpAddress.
If the customer is online, then the flag customerOnline must be set to true, and the customerIpAddress must be given. This implies that Ibanity will indicate to the bank that the customer is online, and the bank should accept the request.
If the customer is offline, then the flag customerOnline must be set to false, and the customerIpAddress is not required. This implies that Ibanity will indicate to the bank that the customer is offline, and the bank will only accept the request, if the legal number of access per day is not exceeded

**Comment by cleclefibanity** [comment]
> New synchronization attribute: customerOnline true|false (false if not set), and customerIpAddress.
> If the customer is online, then the flag customerOnline must be set to true, and the customerIpAddress must be given. This implies that Ibanity will indicate to the bank that the customer is online, and the bank should accept the request.
> If the customer is offline, then the flag customerOnline must be set to false, and the customerIpAddress is not required. This implies that Ibanity will indicate to the bank that the customer is offline, and the bank will only accept the request, if the legal number of access per day is not exceeded

@damienbr

**Review by damienbr — COMMENTED** [review]
It would be nice if you update SynchronizationServiceImplTest or AccountsServiceImplTest or both :D in order to see if the mapping works as expected for the resource and the meta.

File: src/main/java/com/ibanity/apis/client/products/xs2a/models/read/SynchronizationReadQuery.java [hunk 1]
```diff
@@ -19,6 +19,8 @@
     private String subtype;
     private String resourceId;
     private String resourceType;
+    private boolean customerOnline;
```
Comment (damienbr) [inline src/main/java/com/ibanity/apis/client/products/xs2a/models/read/SynchronizationReadQuery.java:22]: You have to update  SynchronizationServiceImpl:31 in order to pass the new properties in the request.

Comment (cleclefibanity) [reply src/main/java/com/ibanity/apis/client/products/xs2a/models/read/SynchronizationReadQuery.java:22]: split/create SynchronizationCreateQuery from SynchronizationReadQuery and map readonly read/write fields

Comment (damienbr) [reply src/main/java/com/ibanity/apis/client/products/xs2a/models/read/SynchronizationReadQuery.java:22]: Alright but that's a breaking change so you'll have to update the major version.

File: src/test/java/com/ibanity/apis/client/products/xs2a/services/impl/SynchronizationServiceImplTest.java [hunk 2]
```diff
@@ -51,18 +52,19 @@ void setUp() {
 
     @Test
     void create() throws Exception {
-        SynchronizationReadQuery synchronizationReadQuery =
-                SynchronizationReadQuery.builder()
+        SynchronizationCreationQuery synchronizationCreationQuery =
```
Comment (damienbr) [inline src/test/java/com/ibanity/apis/client/products/xs2a/services/impl/SynchronizationServiceImplTest.java:55]: If the customerOnline property is part of the response, I would suggest to update create_synchronization.json in order to return true aswell to make sure the property is properly mapped

File: src/main/java/com/ibanity/apis/client/products/xs2a/models/read/SynchronizationReadQuery.java (hunk 1 above)
Comment (cleclefibanity) [reply src/main/java/com/ibanity/apis/client/products/xs2a/models/read/SynchronizationReadQuery.java:22]: agreed. How do we handle versioning here?

File: src/test/java/com/ibanity/apis/client/products/xs2a/services/impl/SynchronizationServiceImplTest.java (hunk 2 above)
Comment (cleclefibanity) [reply src/test/java/com/ibanity/apis/client/products/xs2a/services/impl/SynchronizationServiceImplTest.java:55]: solved

**Pull request merged**